### PR TITLE
fix: iterator/stat utility edge cases and add tests

### PR DESCRIPTION
**Pull request by dasepmoch**
## Description

This PR fixes correctness and safety issues discovered while expanding unit coverage for the round-robin iterator, statistics tracker, and utility helpers.

- Compare round-robin iterators by position instead of the value at that position, preventing duplicate values at different positions from comparing equal.
- Implement the positional comparison required by the round-robin iterator's relational operators.
- Constrain the round-robin iterator and its factory to random-access iterators, matching the operations and iterator category they expose.
- Initialize the maximum statistic with `std::numeric_limits<T>::lowest()`, allowing negative floating-point samples to be tracked correctly.
- Handle empty ranges in `hex_vec()` without decrementing an invalid end iterator.
- Add regression coverage for the corrected behavior and broader coverage for the existing round-robin, statistics, and utility APIs.
- Normalize include-section comments and direct includes across the test sources to match project style.

## Type of Change

- [ ] **feat**: New feature (non-breaking change which adds functionality)
- [x] **fix**: Bug fix (non-breaking change which fixes an issue)
- [ ] **docs**: Documentation only changes
- [x] **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semicolons, etc.)
- [ ] **refactor**: Code change that neither fixes a bug nor adds a feature
- [ ] **perf**: Code change that improves performance
- [x] **test**: Adding missing tests or correcting existing tests
- [ ] **build**: Changes that affect the build system or external dependencies
- [ ] **ci**: Changes to CI configuration files and scripts
- [ ] **chore**: Other changes that don't modify src or test files
- [ ] **revert**: Reverts a previous commit
- [ ] **BREAKING CHANGE**: Introduces a breaking change (can be combined with any type above)

## Checklist

- [x] Code follows the style guidelines of this project
- [x] Code has been self-reviewed
- [x] Code has been commented, particularly in hard-to-understand areas
- [x] Code docstring/documentation-blocks for new or existing methods/components have been added or updated
- [x] Unit tests have been added or updated for any new or modified functionality

### AI Usage

See our [AI usage policy](https://docs.lizardbyte.dev/latest/developers/contributing.html#ai-usage).

- [ ] **None**: No AI tools were used in creating this PR
- [ ] **Light**: AI provided minor assistance (formatting, simple suggestions)
- [ ] **Moderate**: AI helped with code generation or debugging specific parts
- [x] **Heavy**: AI generated most or all of the code changes

### CODE REVIEW

File: tests/unit/test_round_robin.cpp [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+ * @file tests/unit/test_round_robin.cpp
+ * @brief Test src/round_robin.h.
+ */
+#include "../tests_common.h"
+
+#include <src/round_robin.h>
+
+#include <vector>
+
+// ========== Basic iteration tests ==========
```
Comment (ReenigneArcher) [inline tests/unit/test_round_robin.cpp]: Not really a fan of this style of comment. Can you change them to doxygen style doc blocks?

Or like

```
//
// Basic iteration tests
//
```

File: tests/unit/test_stat_trackers.cpp [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+#include <src/stat_trackers.h>
+
+#include <thread>
+
+// ========== Format helper tests ==========
+
+TEST(StatTrackersFormatTests, OneDigitAfterDecimal) {
+  auto fmt = stat_trackers::one_digit_after_decimal();
+  std::string result = (fmt % 3.14159).str();
```
Comment (ReenigneArcher) [inline tests/unit/test_stat_trackers.cpp]: If you don't want to fix this sonar warning, you can use another number that isn't "pi".

**Review by ReenigneArcher — CHANGES_REQUESTED** [review]
This is awesome! Always glad to have more tests added!

Looks pretty good, but could you fix the sonar reported sonar issues as well as adjust the style of the comments mentioned below.

Also, please update the PR to use the correct template. You can find it at https://github.com/LizardByte/.github/blob/master/.github/pull_request_template.md?plain=1

**Review by ReenigneArcher — DISMISSED** [review]

**Comment by ReenigneArcher** [comment]
Thanks for the PR! While reviewing the changes I discovered some edge cases in the src code that needed to be fixed, so the tests are already proving to be useful.

I've pushed additional changes to this PR to address those edge cases, as well as a small style change for the tests.
